### PR TITLE
[fix] Fix unrecognized column name delete handler #32404

### PR DESCRIPTION
**Pull request by gavinchou**
branch-2.0 #32404

### CODE REVIEW

File: regression-test/suites/delete_p0/test_delete_handler.groovy [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    // test condition operator
+    sql """drop table if exists td1;"""
+    sql """
+    CREATE TABLE `td1` (
+      `id` int(11) NULL,
+      `name` varchar(255) NULL,
+      `score` int(11) NULL
+    ) ENGINE=OLAP
+    COMMENT 'OLAP'
+    DISTRIBUTED BY HASH(`id`) BUCKETS 1;
```
Comment (gavinchou) [inline regression-test/suites/delete_p0/test_delete_handler.groovy:31]:
```suggestion
    DISTRIBUTED BY HASH(`id`) BUCKETS 1
    PROPERTIES ( "replication_num" = "1" );
```

File: regression-test/suites/delete_p0/test_delete_handler.groovy [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      `id` int(11) NULL,
+      `name` varchar(255) NULL,
+      `@score` int(11) NULL,
+      `scoreIS` int(11) NULL,
+      `sc ore` int(11) NULL,
+      `score IS score` int(11) NULL,
+      `_a-zA-Z0-9_.+-/?@#\$%^&*" ,:` int(11) NULL
+    ) ENGINE=OLAP
+    COMMENT 'OLAP'
+    DISTRIBUTED BY HASH(`id`) BUCKETS 1;
```
Comment (gavinchou) [inline regression-test/suites/delete_p0/test_delete_handler.groovy:61]:
```suggestion
    DISTRIBUTED BY HASH(`id`) BUCKETS 1
    PROPERTIES ( "replication_num" = "1" );
```

**Comment by gavinchou** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]

**Pull request merged**